openedx/core/djangoapps/content/search/documents.py

```diff
@@ -340,11 +340,21 @@ def _collections_for_content_object(object_id: UsageKey | LearningContextKey) ->
     # Gather the collections associated with this object
     collections = None
     try:
-        component = lib_api.get_component_from_usage_key(object_id)
-        collections = authoring_api.get_entity_collections(
-            component.learning_package_id,
-            component.key,
-        )
+        if isinstance(object_id, UsageKey):
+            component = lib_api.get_component_from_usage_key(object_id)
+            collections = authoring_api.get_entity_collections(
+                component.learning_package_id,
+                component.key,
+            )
+        elif isinstance(object_id, LibraryContainerLocator):
+            container = lib_api.get_container_from_key(object_id)
+            collections = authoring_api.get_entity_collections(
+                container.publishable_entity.learning_package_id,
+                container.key,
+            )
+        else:
+            return result
+
     except ObjectDoesNotExist:
         log.warning(f"No component found for {object_id}")
```

Comment:
nit: we might get ObjectNotFound for the container or component here:

```suggestion
        log.warning(f"No library item found for {object_id}")
```